app/models/user_notification.rb

```diff
@@ -9,4 +9,7 @@ class UserNotification < ApplicationRecord
 
   belongs_to :activity, inverse_of: :user_notifications
   belongs_to :user, inverse_of: :notifications
+
+  scope :ordered_by_updated_at, -> { order(updated_at: :asc) }
+  scope :popup, -> { where(notification_type: notification_types[:popup]) }
```

Comment:
Can you check if this is necessary? I thought such enum scopes are added by default.

Reply:
Fixed!